core/crypto/primitives/aes.go

```diff
@@ -39,108 +38,101 @@ func GenAESKey() ([]byte, error) {
 	return GetRandomBytes(AESKeyLength)
 }
 
-// PKCS7Padding pads as prescribed by the PKCS7 standard
-func PKCS7Padding(src []byte) []byte {
-	padding := aes.BlockSize - len(src)%aes.BlockSize
-	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
-	return append(src, padtext...)
+// PKCS7Padding adds padding for any block smaller than AES block size
+func PKCS7Padding(message []byte) []byte {
+	if len(message)%aes.BlockSize == 0 {
```

Comment:
OK, that's the fundamental change here. Nice! Let's add a few tests, padding source/messages of various sizes.